### DISCUSSION TITLE
Passing a javascript function into a c# method as an IntPtr

### DISCUSSION THREAD

**Discussion by LaptopCat** [vfsfitvnm/frida-il2cpp-bridge#473] [Q&A] [answered]
I am trying to use the BestHTTP library with [this example code](https://benedicht.github.io/BestHTTP-Documentation/pages/best_http2/protocols/http/GettingStarted.html), but it requires passing in a function as a callback

The constructor method and the callback class look like this:
![image](https://github.com/vfsfitvnm/frida-il2cpp-bridge/assets/98688203/b25c2010-46dd-4ecc-9779-cb5876ae9a8a)
![image](https://github.com/vfsfitvnm/frida-il2cpp-bridge/assets/98688203/c38808cc-097e-4818-bcb9-c93c7bd43138)

Here is my current code:
```ts
const System = Il2Cpp.domain.assembly("System").image
const Game = Il2Cpp.domain.assembly("Assembly-CSharp").image

/* ... */

const uri = System.class("System.Uri").alloc()
uri.method(".ctor").invoke(Il2Cpp.string("https://google.com"))

const req = Game.class("BestHTTP.HTTPRequest").alloc()
req.method(".ctor").overload("System.Uri", "BestHTTP.OnRequestFinishedDelegate").invoke(uri, /* ??? */)
req.method("Send").invoke()
```

Sending a request without the overload and callback worked for me, but I need some kind of way to wait until a response is given and using the callback looks the most effective

**Comment by vfsfitvnm** [comment, marked as answer]
`Il2Cpp::delegate`: https://github.com/vfsfitvnm/frida-il2cpp-bridge/wiki/Changelog#v080

**Reply by LaptopCat** [reply]
Thanks! It worked
